crates/phactory/api/src/blocks.rs

```diff
@@ -157,3 +157,16 @@ pub mod compat {
         }
     }
 }
+
+pub fn find_scheduled_change(
+    header: &BlockHeader,
+) -> Option<ScheduledChange<u32>> {
+    let filter_log = |log: ConsensusLog<u32>| match log {
+        ConsensusLog::ScheduledChange(change) => Some(change),
+        _ => None,
+    };
+
+    // find the first consensus digest with the right ID which converts to
+    // the right kind of consensus log.
+    header.digest.convert_first(|l| l.consensus_try_to(&GRANDPA_ENGINE_ID).and_then(filter_log))
```

Comment:
Would this verify the changes are authorized by the old validators?

Reply:
Basically it's same data as reading from state. If the header.digest was hacked/broken, the hash would change.

https://spec.polkadot.network/chap-state#defn-block-header

The block hash is hash of (parent_hash + number + state_root + extrinsics_root + digest). Old authority set + justification proves the header hash, header hash proves header.digest

Reply:
Got it.